redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -335,5 +342,297 @@ inline void
+                                         << "event object dbus name is empty";
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    // Get event properties and fill into status conditions
+                    crow::connections::systemBus->async_method_call(
+                        [aResp, hwStatusEventObj](
+                            const boost::system::error_code ec,
+                            const HwStausEventPropertiesType& properties) {
+                            if (ec)
+                            {
+                                BMCWEB_LOG_ERROR
+                                    << "DBUS response error [" << ec.value()
+                                    << " : " << ec.message() << "] when tried "
+                                    << "to get the hardware status event "
+                                    << "object properties " << hwStatusEventObj;
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+
+                            // Event is exist and that will get created when
+                            // the respective hardware is not functional so
+                            // set the state as "Disabled".
+                            aResp->res.jsonValue["Status"]["State"] =
+                                "Disabled";
+
+                            nlohmann::json& conditions =
+                                aResp->res.jsonValue["Status"]["Conditions"];
+                            conditions = nlohmann::json::array();
+                            conditions.push_back(nlohmann::json::object());
+                            nlohmann::json& condition = conditions.back();
+
+                            for (const auto& property : properties)
+                            {
+                                if (property.first == "Associations")
+                                {
+                                    const AssociationsValType* associations =
+                                        std::get_if<AssociationsValType>(
+                                            &property.second);
+                                    if (associations == nullptr)
+                                    {
+                                        BMCWEB_LOG_ERROR
+                                            << "Failed to get the Associations "
+                                            << "from object: "
+                                            << hwStatusEventObj;
+                                        messages::internalError(aResp->res);
+                                        break;
+                                    }
+
+                                    for (auto& assoc : *associations)
+                                    {
+                                        if (std::get<0>(assoc) == "error_log")
```

Comment:
Doesn't this throw?

Reply:
Assuming this is clarified so marking as resolved.